### ISSUE TITLE
Type interfaces from useNuxtApp are not properly resolved in a mono repository with layers

### ISSUE THREAD

**Issue by adamdehaven** [nuxt/nuxt#27693]
### Environment

- Operating System: Darwin
- Node Version:     v18.18.0
- Nuxt Version:     3.12.2
- CLI Version:      3.12.0
- Nitro Version:    2.9.6
- Package Manager:  pnpm@9.3.0
- Builder:          -
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

[GitHub Reproduction](https://github.com/adamdehaven/nuxt-monorepo-reproductions/tree/chore/usenuxtapp-types)

(Ensure you are on the `chore/usenuxtapp-types` branch)

### Describe the bug

The type interface(s) for helpers provided via `plugins` are not automatically/properly resolved within a mono repository that also utilizes Nuxt Layers.

![image](https://github.com/nuxt/nuxt/assets/2229946/092d7c33-dab1-4daa-94b1-fb67eb895944)

## Steps to reproduce

1. Clone [the linked reproduction](https://github.com/adamdehaven/nuxt-monorepo-reproductions/tree/chore/usenuxtapp-types)
2. Install dependencies with `pnpm install`
3. Run `prepare` with `pnpm -r prepare`
4. Open the `layers/core/components/HelloThere.vue` file and notice that the `$hello` helper provided by `layers/core/plugins/hello.ts` does not infer its type.

The type of the simple plugin `provide` should automatically be resolved.

### Advanced use-cases 

In addition, manually declaring the types of the plugin _also does not work_ as [outlined by the docs](https://nuxt.com/docs/guide/directory-structure/plugins#typing-plugins) (and shown below) without adding the additional interface shown in the middle.

Ideally, the types of the `provide` would be inferred/automatically generated.

```typescript
declare module '#app' {
  interface NuxtApp {
    $hello (msg: string): string
  }
}

// This interface appears to be **required** for mono repo + layers support
declare module 'nuxt/app' {
  interface NuxtApp {
    $hello (msg: string): string
  }
}

declare module 'vue' {
  interface ComponentCustomProperties {
    $hello (msg: string): string
  }
}

export {}
```

### Additional context

Even when external modules provide the proper interfaces ([as seen here with `nuxt-open-fetch`](https://github.com/enkot/nuxt-open-fetch/blob/797c37405936c655c25870a9acea93e3c99b980e/src/module.ts#L202-L212)) the interfaces for the global helpers are not properly resolved in the same repo setup.

**Comment by adamdehaven** [comment]
I got a little further on this, although still not sure of the root cause or if this is related or just _seems related_;I outlined more details [here](https://github.com/enkot/nuxt-open-fetch/pull/47#issuecomment-2181949405)

If a Nuxt app extends a layer, only some of the `paths` from the layer's `tsconfig.json` are making it through to the app's resolved `.nuxt/tsconfig.json` file.

**Comment by adamdehaven** [comment]
Closing as this appears to be resolved with `nuxt@3.13.2` and `vue-tsc@2.1.8`
